junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
> If calling `Thread.interrupt()` on an already interrupted thread is a no-op

This is not always true. If `Thread.interrupt()` is called and the thread clears its interrupt status (by throwing an `InterruptedException` or by calling `Thread.interrupted()` ), next call will set again the interrupt status flag. This will make the next `await` call on any latch to throw an `InterruptedException` with no apparent reason, even in another test.
This is why I added the `testThreadInterrupted` field in the first place.

From my point of view, I think we should remove completely the whole use of interrupts. They can pass from one test to another.